caffe2/ideep/operators/operator_fallback_ideep.h

```diff
@@ -133,7 +138,12 @@ class IDEEPFallbackOp final : public IDEEPOperator {
         if (dtensor->get_dims() != dst_dims) {
           dtensor->resize(dst_dims, itensor::data_type::f32);
         }
-        dtensor->set_data_handle(const_cast<void*>(src.raw_data()));
+        if (output_inplace_[i]) {
+          dtensor->reorder_from(dst_dims, itensor::data_type::f32,
+                                const_cast<void*>(src.raw_data()));
+        } else {
+          dtensor->set_data_handle(const_cast<void *>(src.raw_data()));
+        }
```

Comment:
Only reuse the buffer when the output is not in-place, otherwise "reorder" operation in the 2nd iteration would have source and destination buffers pointing to the same buffer, which is wrong.